lib/DataLayoutAnalysis/Backend/DLAUpdateModelTypes.cpp

```diff
@@ -480,36 +468,33 @@ fillStructWithRecoveredDLATypeAtOffset(model::Binary &Model,
     // can be injected directly as a field in OriginalStruct at Offset.
     if (TargetOriginalFieldIt == OriginalFieldsEnd) {
       revng_assert(Offset == NewOffset);
-      OriginalStruct->Fields()[Offset].Type() = RecoveredType;
+      OriginalStruct.Fields()[Offset].Type() = RecoveredType.copy();
       rc_return;
     }
 
     // Otherwise we have to insert RecoveredType in *TargetOriginalFieldIt at
     // NewOffset.
-    model::QualifiedType OriginalType = TargetOriginalFieldIt->Type();
-    model::TypeDefinition *TargetFieldType = peelConstAndTypedefs(OriginalType)
-                                               .UnqualifiedType()
-                                               .get();
-    auto *TargetFieldStruct = cast<model::StructDefinition>(TargetFieldType);
-
-    uint64_t OldFieldSize = *TargetFieldStruct->size();
+    model::UpcastableType &OriginalType = TargetOriginalFieldIt->Type();
+    auto *Definition = OriginalType->skipConstAndTypedefs()->skipToDefinition();
```

Comment:
Yep, `skipToDefinition` is semantically equivalent to old `UnqualifiedType`.

Reply:
First of all, we decided to get rid of `asDefinition` to reduce confusion around its unnatural typedef handling.

Second of all, even while it was here, `asDefinition` was _asserting_ that the type is a definition (as if not a pointer, array, primitive, etc).

`skipToDefinition`, on the other hand, is for the cases where we care about how separate definitions relate to one another, which is the logic needed in this case.

If you think this is confusing, we can wrap it into a helper, something like `skipToNonTypedefDefinition()` and call it a day.

Reply:
> First of all, we decided to get rid of asDefinition to reduce confusion around its unnatural typedef handling.

So we have `asStructDefinition` but not `asDefinition` (or whatever prefix we choose)? Weird, but OK.

Reply:
Yep, but there are a couple of places that still needed the "non-typedef-skipping" `asDefinition`, so I decided to keep the behavior under an intentionally unrelated name of `tryGetAsDefinition` (since it's relatively long, regex-renaming it into anything shorter is going to be trivial).